src/entities/database.js

```diff
@@ -284,6 +284,43 @@ class DB implements Database {
     )
   }
 
+  async queryLoosePendingTxs() {
+    const snapshotsTable = 'pending_snapshots'
+    const query = Q.sql.select().from(snapshotsTable)
+      .field('tx_hash')
+      .where('block_height = ?', Q.sql.select().from(snapshotsTable)
+        .field('MAX(block_height)'))
+      .union(Q.sql.select().from('txs')
+        .field('hash')
+        .where('tx_state = ?', TX_STATUS.TX_PENDING_STATUS)
+        .where('NOT EXISTS ?', Q.sql.select().from(snapshotsTable)
+          .field('1')
+          .where('tx_hash = hash')))
+      .toString()
+    const dbRes = await this.getConn().query(query)
+    this.#logger.debug('queryLoosePendingTxs:', query, dbRes)
+    return _.map(dbRes.rows, 'tx_hash')
+  }
+
+  async storeNewPendingSnapshot(block: Block) {
+    const loosePendingSet = await this.queryLoosePendingTxs()
+    const txHashes = _.map(block.txs, 'id')
+    const nextSnapshot = loosePendingSet.filter(hash => !txHashes.includes(hash))
+    if (_.isEmpty(nextSnapshot)) {
+      this.#logger.debug('storeNewPendingSnapshot: No new loose txs.')
```

Comment:
`'No new loose txs.'` => `'No pending txs to snapshot.'`

Because a new snapshot can be created even if there are no new **loose** txs, in case there were some txs in the last snapshot and new block confirmed only some of them.